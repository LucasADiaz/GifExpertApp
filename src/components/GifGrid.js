import React from 'react'
import { GifGridItem } from './GifGridItem';
import { getGifts } from '../helpers/getGifts';
import { useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {

    //costumize Hook
    const {data:images, loading} = useFetchGifs(category);

   

    return (
        <>
        <h3 className="animated__animated animate_fadeIn"> {category} </h3>
        {loading && <p className="animated__animated animate_flash">loading</p>}
        <div className="card-grid">
                {
                    images.map((img) => {
                        return <GifGridItem 
                            key={img.id} 
                            {...img}
                        /> 
                    })
                }
        </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}