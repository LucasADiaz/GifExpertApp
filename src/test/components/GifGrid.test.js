import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');


describe('pruebas en el componente GifGrid', () => {
    
    const category = 'One Punch'
    
    test('should  mostrarse correctamente', () => {
        
        useFetchGifs.mockReturnValue({
            data:[],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={category} />)
        expect( wrapper ).toMatchSnapshot();
    });

    test('should debe regresar la informacion | debe mostrar item cuando se cargan imagenes | useFetchGifs', () => {
        
        const gifs = [{
            id:'ABC',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'cualquier cosa'
        }]

        useFetchGifs.mockReturnValue({
            data:[],
            loading: true
        })
        const wrapper = shallow(<GifGrid category={category} />)
        expect( wrapper ).toMatchSnapshot();
    })
    
    
});
