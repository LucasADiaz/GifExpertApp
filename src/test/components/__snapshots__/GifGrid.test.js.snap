// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el componente GifGrid should  mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animated__animated animate_fadeIn"
  >
     
    One Punch
     
  </h3>
  <p
    className="animated__animated animate_flash"
  >
    loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`pruebas en el componente GifGrid should debe regresar la informacion | debe mostrar item cuando se cargan imagenes | useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animated__animated animate_fadeIn"
  >
     
    One Punch
     
  </h3>
  <p
    className="animated__animated animate_flash"
  >
    loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
